src/handlers.rs

```diff
@@ -1,36 +1,243 @@
 // Copyright 2018 Ryan Dahl <ry@tinyclouds.org>
 // All rights reserved. MIT License.
 extern crate libc;
+#[macro_use]
+extern crate log;
 extern crate url;
 
 use libc::c_char;
+use libc::uint32_t;
 use std::ffi::CStr;
+use std::ffi::CString;
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
 use url::Url;
 
-fn string_from_ptr(ptr: *const c_char) -> String {
-    let cstr = unsafe { CStr::from_ptr(ptr as *const i8) };
-    String::from(cstr.to_str().unwrap())
+// TODO(ry) Share this with the def in src/main.rs.
+#[repr(C)]
+pub struct DenoC {
+    _unused: [u8; 0],
+}
+
+// TODO(ry) Share this extern block with those in main.rs.
+// See src/reply.h
+extern "C" {
+    pub fn deno_reply_error(d: *const DenoC, cmd_id: uint32_t, msg: *const c_char);
+    pub fn deno_reply_null(d: *const DenoC, cmd_id: uint32_t);
+    pub fn deno_reply_code_fetch(
+        d: *const DenoC,
+        cmd_id: uint32_t,
+        module_name: *const c_char,
+        filename: *const c_char,
+        source_code: *const c_char,
+        output_code: *const c_char,
+    );
 }
 
+// TODO(ry) SRC_DIR is just a placeholder for future caching functionality.
+static SRC_DIR: &str = "/Users/rld/.deno/src/";
+const ASSET_PREFIX: &str = "/$asset$/";
```

Comment:
Let's keep it for now since there's other code to port that also uses that nomenclature. I'm fine with renaming it all one we have better test coverage.